pkg/config/store.go

```diff
@@ -0,0 +1,107 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ChangeType denotes the type of change
+type ChangeType int
+
+const (
+	// ChangeTypeUpdate - change was an update or a create to a key.
+	ChangeTypeUpdate ChangeType = iota
+	// ChangeTypeDelete - key was removed.
+	ChangeTypeDelete
+)
+
+// Change - A change record in the changelog.
```

Comment:
maybe we need some words like "records a mutation in the underlying mixer config"